source/robot_model/src/Model.cpp

```diff
@@ -437,27 +499,32 @@ Model::inverse_velocity(const std::vector<state_representation::CartesianTwist>&
   this->solver_.updateLinearConstraintsMatrix(this->constraint_matrix_);
   // solve the QP problem
   this->solver_.solve();
-  Eigen::VectorXd solution = this->solver_.getSolution();
   // extract the solution
-  JointVelocities result(joint_positions.get_name(), joint_positions.get_names());
-  Eigen::VectorXd delta_q = solution.head(nb_joints);
-  double T = solution(nb_joints);
-  result.set_velocities(delta_q / T);
-  return result;
+  JointPositions joint_displacement(joint_positions.get_name(),
+                                    joint_positions.get_names(),
+                                    this->solver_.getSolution().head(nb_joints));
+  double dt = this->solver_.getSolution().tail(1)(0);
+  return JointPositions(joint_displacement) / dt;
 }
 
 state_representation::JointVelocities Model::inverse_velocity(const state_representation::CartesianTwist& cartesian_twist,
                                                               const state_representation::JointPositions& joint_positions,
-                                                              const InverseVelocityParameters& parameters) {
+                                                              const QPInverseVelocityParameters& parameters,
+                                                              const std::string& frame_name) {
+  std::string actual_frame_name = frame_name.empty() ? this->robot_model_.frames.back().name : frame_name;
```

Comment:
I like this, would this be applicable to other functions too?

Reply:
I will answer to myself: we can use this on line 256 as well.

Reply:
yes I tried to reduce as much as possible this check and I think it is worth reusing it somewhere else